app/controllers/course/forum/posts_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Course::Forum::PostsController < Course::Forum::Controller
+  before_action :load_topic
+  authorize_resource :topic
+
+  include Course::Discussion::PostsConcern
+
+  def create
+    if super
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  success: t('course.discussion.posts.create.success')
+    else
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  danger: t('course.discussion.posts.create.failure',
+                            error: @post.errors.full_messages.to_sentence)
+    end
+  end
+
+  def update
+    if super
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  success: t('course.discussion.posts.update.success')
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    if super
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  success: t('course.discussion.posts.destroy.success')
+    else
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  danger: t('course.discussion.posts.destroy.failure',
+                            error: @post.errors.full_messages.to_sentence)
+    end
+  end
+
+  protected
+
+  def create_topic_subscription
+    return true if @topic.subscribed_by?(current_user)
+    @topic.subscribed_by(current_user)
+  end
+
+  def topic
+    @topic.acting_as
```

Comment:
Call `acting_as` in the parent controller ?  because you are `authorize_resource :topic` here, I'm not sure whether `@topic.acting_as` or the actual `@topic` get authorised.